### ISSUE TITLE
[boost-build] build failure for x64-windows but boost-build:x86-windows works

### ISSUE THREAD

**Issue by alinaliBQ** [microsoft/vcpkg#31512]
### Operating system

Windows

### Steps to reproduce the behavior

1. Have vcpkg.json file as below (I think only the boost packages matter in this context but included the original file here for full context).

```
{
  "name": "myProject",
  "version-string": "1.0.0",
  "dependencies": [
    "openssl",
    "boost-test",
    "boost-asio",
    "boost-chrono",
    "boost-interprocess",
    "boost-regex",
    "boost-system",
    "boost-thread",
    "curl"
  ],
  "overrides": [
  ],
  "builtin-baseline": "5bb5f3923a33d862b25c18fd4514e08c74c698e1"
}
```

2. Run command ` C:\vcpkg\vcpkg.exe install --triplet x64-windows --x-install-root="C:/vcpkg/installed"  | Write-Host`

- Without vcpkg.json, one can use `vcpkg install boost-build:x64-windows` in vcpkg classic mode.

### Failure logs

bootstrap-x64-windows-err.log
```
File Not Found
```

bootstrap-x64-windows-out.log

```
Building the B2 engine..
Call_If_Exists "bin\VCVARS32.BAT"
###
### Using 'msvc' toolset.
###

C:\vcpkg\packages\boost-build_x64-windows\tools\boost-build\src\engine>"cl" /nologo /MP /MT /TP /Feb2 /wd4996 /O2 /GL /EHsc   -DNDEBUG  builtins.cpp class.cpp command.cpp compile.cpp constants.cpp cwd.cpp debug.cpp debugger.cpp execcmd.cpp execnt.cpp execunix.cpp filent.cpp filesys.cpp fileunix.cpp frames.cpp function.cpp glob.cpp hash.cpp hcache.cpp hdrmacro.cpp headers.cpp jam.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules.cpp native.cpp object.cpp option.cpp output.cpp parse.cpp pathnt.cpp pathsys.cpp pathunix.cpp regexp.cpp rules.cpp scan.cpp search.cpp jam_strings.cpp startup.cpp subst.cpp sysinfo.cpp timestamp.cpp variable.cpp w32_getreg.cpp modules/order.cpp modules/path.cpp modules/property-set.cpp modules/regex.cpp modules/sequence.cpp modules/set.cpp /link kernel32.lib advapi32.lib user32.lib 
builtins.cpp
class.cpp
command.cpp
compile.cpp
constants.cpp
cwd.cpp
debug.cpp
debugger.cpp
execcmd.cpp
execnt.cpp
execunix.cpp
filent.cpp
filesys.cpp
fileunix.cpp
frames.cpp
function.cpp
glob.cpp
hash.cpp
hcache.cpp
hdrmacro.cpp
headers.cpp
jam.cpp
jamgram.cpp
lists.cpp
make.cpp
make1.cpp
md5.cpp
mem.cpp
modules.cpp
native.cpp
object.cpp
option.cpp
output.cpp
parse.cpp
pathnt.cpp
pathsys.cpp
pathunix.cpp
regexp.cpp
rules.cpp
scan.cpp
search.cpp
jam_strings.cpp
startup.cpp
subst.cpp
sysinfo.cpp
timestamp.cpp
variable.cpp
w32_getreg.cpp
order.cpp
path.cpp
property-set.cpp
regex.cpp
sequence.cpp
set.cpp
LINK : fatal error LNK1104: cannot open file 'b2.exe'

C:\vcpkg\packages\boost-build_x64-windows\tools\boost-build\src\engine>dir *.exe 
 Volume in drive C is BOOTCAMP
 Volume Serial Number is 0246-9FF7

 Directory of C:\vcpkg\packages\boost-build_x64-windows\tools\boost-build\src\engine


Failed to build the B2 engine.
```

Console Output (only console output for classic mode is posted, as it shows same error with manifest mode):
```
PS C:\> vcpkg install boost-build:x64-windows --triplet x64-windows
Computing installation plan...
warning: vcpkg appears to be in a Visual Studio prompt targeting x86 but installing for x64-windows. Consider using --triplet x86-windows or --triplet x86-uwp.
The following packages will be built and installed:
    boost-build[core]:x64-windows -> 1.82.0#1
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\AlinaLi\AppData\Local\vcpkg\archives in 463 us. Use --debug to see more details.
Installing 1/1 boost-build:x64-windows...
Building boost-build[core]:x64-windows...
-- Using cached boostorg-build-boost-1.82.0.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/boost-build/src/ost-1.82.0-e88c66a4cd.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/boostorg-build-boost-1.82.0.tar.gz
-- Applying patch 0002-fix-get-version.patch
-- Applying patch 0003-fix-emscripten.patch
-- Applying patch 0004-workaround-MSVC-codegen-bug-in-17.5-on-ARM64.patch
-- Using source at C:/vcpkg/buildtrees/boost-build/src/ost-1.82.0-e88c66a4cd.clean
-- Using cached boost_LICENSE_1_0.txt.
-- Using cached boost-1.82.0-boostcpp.jam.
-- Installing: C:/vcpkg/packages/boost-build_x64-windows/share/boost-build/copyright
-- Bootstrapping...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: C:/vcpkg/packages/boost-build_x64-windows/tools/boost-build/bootstrap.bat msvc
    Working Directory: C:/vcpkg/packages/boost-build_x64-windows/tools/boost-build
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\boost-build\bootstrap-x64-windows-out.log
      C:\vcpkg\buildtrees\boost-build\bootstrap-x64-windows-err.log

Call Stack (most recent call first):
  ports/boost-build/portfile.cmake:65 (vcpkg_execute_required_process)
  scripts/ports.cmake:147 (include)


error: building boost-build:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+boost-build
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[boost-build]+Build+error&body=Copy+issue+body+from+C%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```

### Additional context

On May-12th, the build for `boost-build:x64-windows` is working with boost on classic mode, but today I tried with manifest mode and started encountering issues, and after some testing I discovered it is not building on classic mode as well. I have tried deleting vcpkg and re-cloning but have the same error. 
- The failure is at the bootstrapping step. 
- vcpkg also warns for Visual Studio targeting x86, but my Visual Studio is x64
- the build for  `boost-build:x86-windows` works without any issues. 

I tried vcpkg from older branches but I now still have the same problem. It might be due to my environment change as boost-build should be building from the same source code as before, but I have no idea what it might be, any help would be appreciated, thanks.

**Comment by Cheney-W** [comment]
I can't reproduce this issue in my side. I think you could repair your VS first, then clone a new `vcpkg` repository and install `boost-build:x64-windows` again.

**Comment by alinaliBQ** [comment]
Hi @Cheney-W , I have tried uninstalling the unused Visual Studio 2019, leaving only Visual Studio 2022. I have also repaired the Visual Studio 2022, and cloned the new `vcpkg`, but the error persists.

**Comment by alinaliBQ** [comment]
@Cheney-W  The version of Visual Studio I was using was 17.6, and I also tried 17.6.1, and both had the same error as above. My co-worker mentioned that VS 17.5.5 worked for him, so I used that VS version instead, and now I can build vcpkg `boost-build`. I believe you might be able to reproduce the issue with the newest version of Visual Studio (17.6.1). I'm not sure if the error is from vcpkg, boost, or Visual Studio, although it does feel like a Visual Studio problem.

**Comment by Cheney-W** [comment]
I'm sorry, but I still couldn't reproduce this issue. I was previously using VS2022 17.5.5, and now I have upgraded to VS2022 17.6.1. Then I re-cloned vcpkg and installed xx, and I was able to install it successfully.

Additionally, what's the results when you try the following steps:
```
1. Open x64 Native Tools Command Prompt for VS 2022
2. Pushd C:\vcpkg\packages\boost-build_x64-windows\tools\boost-build
3. Run bootstrap.bat
```

**Comment by alinaliBQ** [comment]
No worries. I actually never used `x64 Native Tools Command Prompt for VS 2022` when installing vcpkg packages, and I only used PowerShell 7 or DeveloperShell for VS 2022, that might be the cause of the errors... I'll try when I have time, for now I'll use VS 17.5.5. In the main time I have some questions:
1. Since [vcpkg is included in VS 2022 Ver 17.6](https://devblogs.microsoft.com/cppblog/vcpkg-is-now-included-with-visual-studio/), did you encounter any conflicts when re-cloning vcpkg? 
2. Is `x64 Native Tools Command Prompt for VS 2022` the recommended tool for running vcpkg?

**Comment by Cheney-W** [comment]
Thanks for your reply!

> 1. Since [vcpkg is included in VS 2022 Ver 17.6](https://devblogs.microsoft.com/cppblog/vcpkg-is-now-included-with-visual-studio/), did you encounter any conflicts when re-cloning and using vcpkg?

In general, conflicts are not expected to occur. The typical approach we use for vcpkg is to clone a vcpkg repository from GitHub to a local directory, such as `E:\vcpkg`. The files in this vcpkg folder are mostly identical to the ones on GitHub. Then, we open a regular PowerShell window in this folder and execute vcpkg-related commands.

The vcpkg that comes with 17.6 is installed in the `${VCINSTALLDIR}\vcpkg` directory and does not require running bootstrap.bat. It already includes a fixed version of vcpkg.exe, which is automatically recognized only when you open Developer PowerShell for VS 2022. It won't be recognized in a regular PowerShell window.

When I mentioned re-cloning a vcpkg, I was referring to the general usage approach, where the installation commands are executed within this vcpkg folder. It is unrelated to the vcpkg installed through Visual Studio.

> 2\. Is `x64 Native Tools Command Prompt for VS 2022` the recommended tool for running vcpkg?

No, that's not the case.
After executing the `vcpkg install` command, `vcpkg.exe` runs the appropriate `vcvarsall.bat` script based on the triplet. Since you installed the `x64-windows` triplet, it means that the compilation environment uses `vcvarsall_bat amd64`, which is the same as the one used by the `x64 Native Tools Command Prompt for VS 2022`. So, I suggest that you use that command prompt for testing.

> 3\. Which Windows platform are you using? Mine is Windows 10 on Intel mac using Windows bootcamp.

Mine:
![image](https://github.com/microsoft/vcpkg/assets/38240633/971039d8-06ae-4275-ae1e-b0d203aaae67)

**Comment by Cheney-W** [comment]
There are currently some issues with the vcpkg installed through the VS Installer. Please refer to: https://github.com/microsoft/vcpkg/issues/31565.

**Comment by dg0yt** [comment]
#31565 is a different error.

Here:
> ~~~
>     See logs for more information:
>       C:\vcpkg\buildtrees\boost-build\bootstrap-x64-windows-out.log
>       C:\vcpkg\buildtrees\boost-build\bootstrap-x64-windows-err.log

Nobody interested in these files?

**Comment by PhoebeHui** [comment]
@dg0yt, the user had put the content of two logs to issue report.

Failures:
LINK : fatal error LNK1104: cannot open file 'b2.exe'

@alinaliBQ. could you help maually check if b2 could be generated successfully? may be deleted by anti-software?

1. Open x64 native Tools command:
2. pushd C:\vcpkg\packages\boost-build_x64-windows\tools\boost-build\src\engine
3. Execute the following command and see if b2.exe generated.

```
cl /nologo /MP /MT /TP /Feb2 /wd4996 /O2 /GL /EHsc   -DNDEBUG  builtins.cpp class.cpp command.cpp compile.cpp constants.cpp cwd.cpp debug.cpp debugger.cpp execcmd.cpp execnt.cpp execunix.cpp filent.cpp filesys.cpp fileunix.cpp frames.cpp function.cpp glob.cpp hash.cpp hcache.cpp hdrmacro.cpp headers.cpp jam.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules.cpp native.cpp object.cpp option.cpp output.cpp parse.cpp pathnt.cpp pathsys.cpp pathunix.cpp regexp.cpp rules.cpp scan.cpp search.cpp jam_strings.cpp startup.cpp subst.cpp sysinfo.cpp timestamp.cpp variable.cpp w32_getreg.cpp modules/order.cpp modules/path.cpp modules/property-set.cpp modules/regex.cpp modules/sequence.cpp modules/set.cpp /link kernel32.lib advapi32.lib user32.lib 
```

**Comment by alinaliBQ** [comment]
@PhoebeHui Oh my, you're correct. I just checked the anti-malware and indeed the b2 files were deleted by it.  
With VS 17.6, I remember when building b2 with fresh vcpkg clone with regular Power Shell, the logs would say the build is successful but the b2 file has `access denied` error, and when I run the vcpkg install as administrator for the second time, the `access denied` error is gone, and instead we're back to the same error I originally posted. I previously thought the `access denied` error was due to me not running Power Shell as administrator, but it is very likely anti-malware related. However, it is very strange that I'm not encountering this problem with VS 17.5.5, and the anti-malware is only deleting b2 with VS 17.6, it might be due to the anti-malware settings or something... I'll check with IT for this one.

![image](https://github.com/microsoft/vcpkg/assets/96995091/5acd012c-bb01-49a3-a6fd-2f998699d3d6)

![image](https://github.com/microsoft/vcpkg/assets/96995091/a92b934b-410d-4090-b653-0b0f95c9db08)
